src/traces/pie/index.js

```diff
@@ -373,6 +373,12 @@ pie.plot = function(gd, cdpie) {
                     }
                 }
 
+                function handleClick (evt) {
+                    gd._hoverdata = pt;
+                    gd._hoverdata.trace = slices[0][0].__data__.trace;
```

Comment:
Only the first slice in the array contains the `trace`. I'm not sure if this is something within plotly.js or if it's a d3 thing.